### PR TITLE
Add ECDSA-384 key support for TPM

### PR DESCRIPTION
**Pull request by shrikant1407**
Add ECDSA-384 key support for TPM

Signed-off-by: Shrikant Temburwar <shrikant.temburwar@intel.com>

### CODE REVIEW

File: crypto/openssl/tpm20_Utils.c [hunk 1]
```diff
@@ -202,11 +208,17 @@ int32_t fdo_tpm_get_hmac(const uint8_t *data, size_t data_length, uint8_t *hmac,
...
 	} else {
-
+#if defined(ECDSA256_DA)
 		ret_val = Esys_HMAC_Start(esys_context, hmac_key_handle,
 					  auth_session_handle, ESYS_TR_NONE,
 					  ESYS_TR_NONE, &null_auth,
 					  TPM2_ALG_SHA256, &sequence_handle);
+#else
+		ret_val = Esys_HMAC_Start(esys_context, hmac_key_handle,
+					  auth_session_handle, ESYS_TR_NONE,
```
Comment (tajnisha) [inline crypto/openssl/tpm20_Utils.c:218]: Can we define TPM2_ALG_SHA_TYPE as TPM2_ALG_SHA256 or TPM2_ALG_SHA384 based on "defined(ECDSA256_DA)" at one place( like how u defined PLATFORM_HMAC_SIZE) ? I hope then we can remove the multiple #ifs

File: crypto/openssl/tpm20_Utils.c [hunk 2]
```diff
@@ -594,11 +606,17 @@ static int32_t fdoTPMEsys_auth_session_init(ESYS_CONTEXT *esys_context,
 	int ret = -1;
 	TPMT_SYM_DEF symmetric = {0};
 	symmetric.algorithm = TPM2_ALG_NULL;
-
+#if defined(ECDSA256_DA)
 	TSS2_RC rval = Esys_StartAuthSession(
 	    esys_context, ESYS_TR_NONE, ESYS_TR_NONE, ESYS_TR_NONE,
 	    ESYS_TR_NONE, ESYS_TR_NONE, NULL, TPM2_SE_HMAC, &symmetric,
 	    TPM2_ALG_SHA256, session_handle);
+#else
```
Comment (tajnisha) [inline crypto/openssl/tpm20_Utils.c:614]: Can we define TPM2_ALG_SHA_TYPE as TPM2_ALG_SHA256 or TPM2_ALG_SHA384 based on "defined(ECDSA256_DA)" at one place( like how u defined PLATFORM_HMAC_SIZE) ? I hope then we can remove the multiple #ifs

File: storage/include/storage_al.h [hunk 3]
```diff
@@ -19,7 +19,11 @@
 #include <stddef.h>
 
 // platform HMAC and blob size
-#define PLATFORM_HMAC_SIZE BUFF_SIZE_32_BYTES
+#if defined(DEVICE_TPM20_ENABLED) && defined(ECDSA384_DA)
```
Comment (tajnisha) [inline storage/include/storage_al.h:22]: With TPM 1.2 is the HMAC size always 32 bytes ?

Comment (shrikant1407) [reply storage/include/storage_al.h:22]: yes and currently we are only supporting TPM 2.0

**Review by tajnisha — APPROVED** [review]

File: utils/tpm_make_ready_ecdsa.sh [hunk 4]
```diff
@@ -71,6 +71,9 @@ execute_cmd_on_failure_exit()
 
 parse_args "$@"
 
+curve="nist_p$ecc"
+primary_key_type="ecc$ecc:aes128cfb"
```
Comment (shrikant1407) [inline utils/tpm_make_ready_ecdsa.sh:75]: Update AES bits to 256 for ecc384

**Review by KiranSukhavasi — APPROVED** [review]

**Review by DukeDavis12 — APPROVED** [review]

**Pull request merged**